routes/deductions/deductions.controller.js

```diff
@@ -0,0 +1,50 @@
+const { validationResult } = require('express-validator')
+const { errorArray2ErrorObject } = require('./../../utils.js')
+//const { loginSchema, sinSchema } = require('./../../formSchemas.js')
+//const API = require('../../api')
+
+module.exports = function (app) {
+  // redirect from "/deductions" → "/login/code", you really shouldn't be here (Unless we add a "info" page)
+  app.get('/deductions', (req, res) => res.redirect('/login/code'))
+
+  app.get('/deductions/rrsp', (req, res) => res.render('deductions/rrsp', { data: req.session || {} }))
+  app.post('/deductions/rrsp', postRRSP)
+
+}
+
+//POST functions that handle setting the login data in the session and handle redirecting to the next page or sending an error to the client.
+//Note that this is not the only error validation, see routes defined above.
+const validateRedirect = req => {
+  let redirect = req.body.redirect || null
+  if (!redirect) {
+    throw new Error(`[POST ${req.path}] 'redirect' parameter missing`)
+  }
+  return redirect
+}
```

Comment:
Now that @katedee's redirect-checking middleware is in, you can include it [in the chain of callbacks that are part of the post request](https://github.com/cds-snc/cra-claim-tax-benefits/blob/master/routes/login/login.controller.js#L15), and delete this `validateRedirect` function.

Reply:
Good call